### PR TITLE
executor: migrate test-infra to testify for join_pkg_test.go

### LINKED ISSUE #28592: migrate test-infra to testify for `join_pkg_test.go`

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #28592 

```release-note
None
```

### CODE REVIEW

File: executor/join_pkg_test.go [hunk 1]
```diff
@@ -16,25 +16,19 @@ package executor
...
-
-type pkgTestSuite struct{}
-
-type pkgTestSerialSuite struct{}
-
-func (s *pkgTestSerialSuite) TestJoinExec(c *C) {
-	c.Assert(failpoint.Enable("github.com/pingcap/tidb/executor/testRowContainerSpill", "return(true)"), IsNil)
-	defer func() { c.Assert(failpoint.Disable("github.com/pingcap/tidb/executor/testRowContainerSpill"), IsNil) }()
+func TestJoinExec(t *testing.T) {
+	require.Nil(t, failpoint.Enable("github.com/pingcap/tidb/executor/testRowContainerSpill", "return(true)"))
```
[the remark this answers is not in the document]
Comment (feitian124) [reply executor/join_pkg_test.go]: fixed. thanks

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
@tisonkun PTAL

**Review by tisonkun — APPROVED** [review]

**Pull request merged**